<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Get started • dce</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Get started">
<meta property="og:description" content="dce">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">dce</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/dce.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/pathway_databases.html">Overview of pathway network databases</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/cbg-ethz/dce/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="dce_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Get started</h1>
                        <h4 class="author">Kim Philipp Jablonski, Martin Pirkl</h4>
            
            <h4 class="date">2021-02-05</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/cbg-ethz/dce/blob/master/vignettes/dce.Rmd"><code>vignettes/dce.Rmd</code></a></small>
      <div class="hidden name"><code>dce.Rmd</code></div>

    </div>

    
    
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p>One cause of diseases like cancer is the de-regulation of signalling pathways. The interaction of two or more genes is changed and cell behaviour is changed in the malignant tissue.</p>
<p>The estimation of causal effects from observational data has previously been used to elucidate gene interactions. We extended this notion to compute Differential Causal Effects (DCE). We compare the causal effects in malignant tissue (e.g., from a tumor) and healthy tissue to detect differences in the gene interactions.</p>
<p>However, computing causal effects purely from observational data is difficult, because beforehand the gene network has to be reconstructed. To overcome this issue we use prior knowledge from literature. This largely improves performance and makes the estimation of DCEs more accurate.</p>
<p>Overall we can detect pathways which play a prominent role in tumorigenesis. We can even pinpoint specific interaction in the pathway that make a large contribution to the rise of the disease.</p>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"dce"</span><span class="op">)</span></code></pre></div>
</div>
<div id="load-required-packages" class="section level1">
<h1 class="hasAnchor">
<a href="#load-required-packages" class="anchor"></a>Load required packages</h1>
<p>Load <code>dce</code> package and other required libraries.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># fix "object 'guide_edge_colourbar' of mode 'function' was not found"</span>
<span class="co"># when building vignettes</span>
<span class="co"># (see also https://github.com/thomasp85/ggraph/issues/75)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggraph.data-imaginist.com">ggraph</a></span><span class="op">)</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">curatedTCGAData</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">TCGAutils</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment">SummarizedExperiment</a></span><span class="op">)</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/">cowplot</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">graph</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/cbg-ethz/dce">dce</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span></code></pre></div>
</div>
<div id="introductory-example" class="section level1">
<h1 class="hasAnchor">
<a href="#introductory-example" class="anchor"></a>Introductory example</h1>
<p>To demonstrate the basic idea of Differential Causal Effects (DCEs), we first create a wild-type network by setting up its adjacency matrix. The specified edge weights correspond to direct causal effects and total causal effects are defined accordingly <span class="citation">(Pearl 2010)</span>. This way, the detected dysregulations are endowed with a causal interpretation and spurious correlations are ignored. This can be achieved by assuming that the underlying network models causal relationships and selecting valid adjustment sets. In a biological setting, these networks correspond, for example, to a KEGG pathway <span class="citation">(Kanehisa et al. 2004)</span> in a healthy cell. Here, the edge weights correspond to proteins facilitating or inhibiting each others expression levels.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">graph_wt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span>, <span class="fl">3</span>, <span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">graph_wt</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">graph_wt</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"B"</span>, <span class="st">"C"</span><span class="op">)</span>
<span class="va">graph_wt</span></code></pre></div>
<pre><code>##   A B C
## A 0 1 1
## B 0 0 1
## C 0 0 0</code></pre>
<p>In case of a disease, these pathways can become dysregulated. This can be expressed by a change in edge weights.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">graph_mt</span> <span class="op">&lt;-</span> <span class="va">graph_wt</span>
<span class="va">graph_mt</span><span class="op">[</span><span class="st">"A"</span>, <span class="st">"B"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">2.5</span> <span class="co"># dysregulation happens here!</span>
<span class="va">graph_mt</span></code></pre></div>
<pre><code>##   A   B C
## A 0 2.5 1
## B 0 0.0 1
## C 0 0.0 0</code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span>
  <span class="fu"><a href="../reference/plot_network.html">plot_network</a></span><span class="op">(</span><span class="va">graph_wt</span>, edgescale_limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">3</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span>,
  <span class="fu"><a href="../reference/plot_network.html">plot_network</a></span><span class="op">(</span><span class="va">graph_mt</span>, edgescale_limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">3</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span>,
  labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"WT"</span>, <span class="st">"MT"</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<p><img src="dce_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<p>By propagating counts which were randomly initialized at root notes, we can generate synthetic expression data for each node in both networks. Both <code>X_wt</code> and <code>X_mt</code> then induce causal effects as defined in their respective adjacency matrices.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">X_wt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simulate_data.html">simulate_data</a></span><span class="op">(</span><span class="va">graph_wt</span><span class="op">)</span>
<span class="va">X_mt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simulate_data.html">simulate_data</a></span><span class="op">(</span><span class="va">graph_mt</span><span class="op">)</span>

<span class="va">X_wt</span> <span class="op">%&gt;%</span>
  <span class="va">head</span></code></pre></div>
<pre><code>##         A   B    C
## [1,] 1117 398  802
## [2,]  964 244  501
## [3,]  963 246  469
## [4,] 1204 502 1032
## [5,]  848 125  251
## [6,] 1163 455  885</code></pre>
<p>Given the network topology (without edge weights!) and expression data from both WT and MT conditions, we can estimate the difference in causal effects for each edge between the two conditions. These are the aforementioned Differential Causal Effects (DCEs).</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dce.html">dce</a></span><span class="op">(</span><span class="va">graph_wt</span>, <span class="va">X_wt</span>, <span class="va">X_mt</span>, solver <span class="op">=</span> <span class="st">"lm"</span><span class="op">)</span>

<span class="va">res</span> <span class="op">%&gt;%</span>
  <span class="va">as.data.frame</span> <span class="op">%&gt;%</span>
  <span class="va">drop_na</span></code></pre></div>
<pre><code>##   source target       dce dce_stderr    dce_pvalue
## 1      A      B 1.5443343 0.03062434 3.373144e-114
## 2      A      C 1.5234426 0.04449333  1.207257e-84
## 3      B      C 0.1152442 0.17882577  5.200452e-01</code></pre>
<p>Visualizing the result shows that we can recover the dysregulation of the edge from <code>A</code> to <code>B</code>. Note that since we are computing total causal effects, the edge from <code>A</code> to <code>C</code> shows the same effect.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Differential Causal Effects between WT and MT condition"</span><span class="op">)</span></code></pre></div>
<p><img src="dce_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
</div>
<div id="application-to-real-data" class="section level1">
<h1 class="hasAnchor">
<a href="#application-to-real-data" class="anchor"></a>Application to real data</h1>
<p>Pathway dysregulations are a common cancer hallmark <span class="citation">(Hanahan and Weinberg 2011)</span>. It is thus of interest to investigate how the causal effect magnitudes in relevant pathways vary between normal and tumor samples.</p>
<div id="retrieve-gene-expression-data" class="section level2">
<h2 class="hasAnchor">
<a href="#retrieve-gene-expression-data" class="anchor"></a>Retrieve gene expression data</h2>
<p>As a showcase, we download breast cancer (BRCA) RNA transcriptomics profiling data from TCGA <span class="citation">(Tomczak, Czerwińska, and Wiznerowicz 2015)</span>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">brca</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/curatedTCGAData/man/curatedTCGAData.html">curatedTCGAData</a></span><span class="op">(</span>
  diseaseCode <span class="op">=</span> <span class="st">"BRCA"</span>,
  assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"RNASeq2*"</span><span class="op">)</span>,
  dry.run <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
<pre><code>## using temporary cache /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//RtmpWtMNNF/BiocFileCache</code></pre>
<pre><code>## snapshotDate(): 2020-10-27</code></pre>
<pre><code>## Working on: BRCA_RNASeq2GeneNorm-20160128</code></pre>
<pre><code>## see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation</code></pre>
<pre><code>## downloading 1 resources</code></pre>
<pre><code>## retrieving 1 resource</code></pre>
<pre><code>## loading from cache</code></pre>
<pre><code>## Working on: BRCA_colData-20160128</code></pre>
<pre><code>## see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation</code></pre>
<pre><code>## downloading 1 resources</code></pre>
<pre><code>## retrieving 1 resource</code></pre>
<pre><code>## loading from cache</code></pre>
<pre><code>## Working on: BRCA_metadata-20160128</code></pre>
<pre><code>## see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation</code></pre>
<pre><code>## downloading 1 resources</code></pre>
<pre><code>## retrieving 1 resource</code></pre>
<pre><code>## loading from cache</code></pre>
<pre><code>## Working on: BRCA_sampleMap-20160128</code></pre>
<pre><code>## see ?curatedTCGAData and browseVignettes('curatedTCGAData') for documentation</code></pre>
<pre><code>## downloading 1 resources</code></pre>
<pre><code>## retrieving 1 resource</code></pre>
<pre><code>## loading from cache</code></pre>
<pre><code>## harmonizing input:
##   removing 13161 sampleMap rows not in names(experiments)
##   removing 5 colData rownames not in sampleMap 'primary'</code></pre>
<p>This will retrieve all available samples for the requested data sets. These samples can be classified according to their site of origin.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/TCGAutils/man/curatedTCGAData-helpers.html">sampleTables</a></span><span class="op">(</span><span class="va">brca</span><span class="op">)</span></code></pre></div>
<pre><code>## $`BRCA_RNASeq2GeneNorm-20160128`
## 
##   01   06   11 
## 1093    7  112</code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sampleTypes</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">Code</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"01"</span>, <span class="st">"06"</span>, <span class="st">"11"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##   Code          Definition Short.Letter.Code
## 1   01 Primary Solid Tumor                TP
## 2   06          Metastatic                TM
## 3   11 Solid Tissue Normal                NT</code></pre>
<p>We can extract Primary Solid Tumor and matched Solid Tissue Normale samples.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># split assays</span>
<span class="va">brca_split</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TCGAutils/man/curatedTCGAData-helpers.html">splitAssays</a></span><span class="op">(</span><span class="va">brca</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"01"</span>, <span class="st">"11"</span><span class="op">)</span><span class="op">)</span>

<span class="co"># only retain matching samples</span>
<span class="va">brca_matched</span> <span class="op">&lt;-</span> <span class="fu">as</span><span class="op">(</span><span class="va">brca_split</span>, <span class="st">"MatchedAssayExperiment"</span><span class="op">)</span>

<span class="va">brca_wt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">brca_matched</span>, <span class="st">"01_BRCA_RNASeq2GeneNorm-20160128"</span><span class="op">)</span>
<span class="va">brca_mt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">brca_matched</span>, <span class="st">"11_BRCA_RNASeq2GeneNorm-20160128"</span><span class="op">)</span></code></pre></div>
</div>
<div id="retrieve-biological-pathway-of-interest" class="section level2">
<h2 class="hasAnchor">
<a href="#retrieve-biological-pathway-of-interest" class="anchor"></a>Retrieve biological pathway of interest</h2>
<p>KEGG <span class="citation">(Kanehisa et al. 2004)</span> provides the breast cancer related pathway <code>hsa05224</code>. It can be easily retrieved using <code>dce</code>.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_pathways.html">get_pathways</a></span><span class="op">(</span>pathway_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>kegg <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Breast cancer"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## INFO [2021-02-05 09:30:08] Processing kegg</code></pre>
<pre><code>## 'select()' returned 1:1 mapping between keys and columns
## 'select()' returned 1:1 mapping between keys and columns
## 'select()' returned 1:1 mapping between keys and columns
## 'select()' returned 1:1 mapping between keys and columns
## 'select()' returned 1:1 mapping between keys and columns
## 'select()' returned 1:1 mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">brca_pathway</span> <span class="op">&lt;-</span> <span class="va">pathways</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">graph</span></code></pre></div>
<p>Luckily it shares all genes with the cancer data set.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">shared_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/graphNEL-class.html">nodes</a></span><span class="op">(</span><span class="va">brca_pathway</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">brca_wt</span><span class="op">)</span><span class="op">)</span>
<span class="fu">glue</span><span class="fu">::</span><span class="fu"><a href="https://glue.tidyverse.org/reference/glue.html">glue</a></span><span class="op">(</span>
  <span class="st">"Covered nodes: {length(shared_genes)}/{length(nodes(brca_pathway))}"</span>
<span class="op">)</span></code></pre></div>
<pre><code>## Covered nodes: 145/145</code></pre>
</div>
<div id="estimate-differential-causal-effects" class="section level2">
<h2 class="hasAnchor">
<a href="#estimate-differential-causal-effects" class="anchor"></a>Estimate Differential Causal Effects</h2>
<p>We can now estimate the differences in causal effects between matched tumor and normal samples on a breast cancer specific pathway.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">dce</span><span class="fu">::</span><span class="fu"><a href="../reference/dce.html">dce</a></span><span class="op">(</span><span class="va">brca_pathway</span>, <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">brca_wt</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">brca_mt</span><span class="op">)</span>, solver <span class="op">=</span> <span class="st">"lm"</span><span class="op">)</span></code></pre></div>
<p>Interpretations may now begin.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res</span> <span class="op">%&gt;%</span>
  <span class="va">as.data.frame</span> <span class="op">%&gt;%</span>
  <span class="va">drop_na</span> <span class="op">%&gt;%</span>
  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">dce</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="va">head</span></code></pre></div>
<pre><code>##   source target        dce dce_stderr   dce_pvalue
## 1  WNT8A   FZD4 -4342.2001  2876.9003 1.326486e-01
## 2   FGF3  FGFR1 -1553.3334  1322.3358 2.413890e-01
## 3  WNT8B   FZD4 -1540.1078   235.1719 4.038543e-10
## 4  WNT7A   FZD4 -1334.3963   598.5797 2.680676e-02
## 5  FGF21  FGFR1  -917.1445  1128.4631 4.172471e-01
## 6  WNT8A   FZD7   817.6733   954.3545 3.924979e-01</code></pre>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">res</span>, nodesize <span class="op">=</span> <span class="fl">20</span>, labelsize <span class="op">=</span> <span class="fl">1</span>, use_symlog <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="dce_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
</div>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Catalina 10.15.7
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] org.Hs.eg.db_3.12.0         AnnotationDbi_1.52.0       
##  [3] dce_0.99.0                  graph_1.68.0               
##  [5] cowplot_1.1.1               forcats_0.5.1              
##  [7] stringr_1.4.0               dplyr_1.0.4                
##  [9] purrr_0.3.4                 readr_1.4.0                
## [11] tidyr_1.1.2                 tibble_3.0.6               
## [13] tidyverse_1.3.0             TCGAutils_1.10.0           
## [15] curatedTCGAData_1.12.0      MultiAssayExperiment_1.16.0
## [17] SummarizedExperiment_1.20.0 Biobase_2.50.0             
## [19] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        
## [21] IRanges_2.24.1              S4Vectors_0.28.1           
## [23] BiocGenerics_0.36.0         MatrixGenerics_1.2.1       
## [25] matrixStats_0.58.0          ggraph_2.0.4               
## [27] ggplot2_3.3.3               BiocStyle_2.18.1           
## 
## loaded via a namespace (and not attached):
##   [1] rappdirs_0.3.3                rtracklayer_1.50.0           
##   [3] prabclus_2.3-2                ragg_0.4.1                   
##   [5] bit64_4.0.5                   knitr_1.31                   
##   [7] multcomp_1.4-15               DelayedArray_0.16.1          
##   [9] data.table_1.13.6             RCurl_1.98-1.2               
##  [11] generics_0.1.0                metap_1.4                    
##  [13] GenomicFeatures_1.42.1        TH.data_1.0-10               
##  [15] RSQLite_2.2.3                 CombinePValue_1.0            
##  [17] bit_4.0.4                     mutoss_0.1-12                
##  [19] xml2_1.3.2                    lubridate_1.7.9.2            
##  [21] httpuv_1.5.5                  assertthat_0.2.1             
##  [23] viridis_0.5.1                 amap_0.8-18                  
##  [25] xfun_0.20                     hms_1.0.0                    
##  [27] evaluate_0.14                 promises_1.1.1               
##  [29] DEoptimR_1.0-8                progress_1.2.2               
##  [31] dbplyr_2.1.0                  readxl_1.3.1                 
##  [33] Rgraphviz_2.34.0              igraph_1.2.6                 
##  [35] DBI_1.1.1                     tmvnsim_1.0-2                
##  [37] apcluster_1.4.8               RcppArmadillo_0.10.1.2.2     
##  [39] ellipsis_0.3.1                backports_1.2.1              
##  [41] bookdown_0.21                 permute_0.9-5                
##  [43] gbRd_0.4-11                   harmonicmeanp_3.0            
##  [45] biomaRt_2.46.2                vctrs_0.3.6                  
##  [47] abind_1.4-5                   Linnorm_2.14.0               
##  [49] cachem_1.0.3                  RcppEigen_0.3.3.9.1          
##  [51] withr_2.4.1                   sfsmisc_1.1-8                
##  [53] ggforce_0.3.2                 robustbase_0.93-7            
##  [55] bdsmatrix_1.3-4               checkmate_2.0.0              
##  [57] vegan_2.5-7                   GenomicAlignments_1.26.0     
##  [59] pcalg_2.7-1                   prettyunits_1.1.1            
##  [61] mclust_5.4.7                  mnormt_2.0.2                 
##  [63] cluster_2.1.0                 ExperimentHub_1.16.0         
##  [65] GenomicDataCommons_1.14.0     crayon_1.4.0                 
##  [67] ellipse_0.4.2                 labeling_0.4.2               
##  [69] FMStable_0.1-2                edgeR_3.32.1                 
##  [71] pkgconfig_2.0.3               tweenr_1.0.1                 
##  [73] nlme_3.1-149                  ggm_2.5                      
##  [75] nnet_7.3-14                   rlang_0.4.10                 
##  [77] diptest_0.75-7                lifecycle_0.2.0              
##  [79] sandwich_3.0-0                BiocFileCache_1.14.0         
##  [81] mathjaxr_1.2-0                modelr_0.1.8                 
##  [83] AnnotationHub_2.22.0          cellranger_1.1.0             
##  [85] rprojroot_2.0.2               polyclip_1.10-0              
##  [87] Matrix_1.2-18                 zoo_1.8-8                    
##  [89] reprex_1.0.0                  viridisLite_0.3.0            
##  [91] bitops_1.0-6                  Biostrings_2.58.0            
##  [93] blob_1.2.1                    scales_1.1.1                 
##  [95] plyr_1.8.6                    memoise_2.0.0                
##  [97] graphite_1.36.0               magrittr_2.0.1               
##  [99] gdata_2.18.0                  zlibbioc_1.36.0              
## [101] compiler_4.0.3                clue_0.3-58                  
## [103] plotrix_3.8-1                 Rsamtools_2.6.0              
## [105] cli_2.3.0                     XVector_0.30.0               
## [107] ps_1.5.0                      MASS_7.3-53                  
## [109] mgcv_1.8-33                   tidyselect_1.1.0             
## [111] stringi_1.5.3                 textshaping_0.2.1            
## [113] highr_0.8                     yaml_2.2.1                   
## [115] askpass_1.1                   locfit_1.5-9.4               
## [117] ggrepel_0.9.1                 grid_4.0.3                   
## [119] tools_4.0.3                   rstudioapi_0.13              
## [121] snowfall_1.84-6.1             gridExtra_2.3                
## [123] farver_2.0.3                  Rtsne_0.15                   
## [125] digest_0.6.27                 BiocManager_1.30.10          
## [127] flexclust_1.4-0               shiny_1.6.0                  
## [129] mnem_1.6.5                    fpc_2.2-9                    
## [131] ppcor_1.1                     Rcpp_1.0.6                   
## [133] broom_0.7.4                   BiocVersion_3.12.0           
## [135] later_1.1.0.1                 httr_1.4.2                   
## [137] ggdendro_0.1.22               kernlab_0.9-29               
## [139] naturalsort_0.1.3             Rdpack_2.1                   
## [141] colorspace_2.0-0              rvest_0.3.6                  
## [143] XML_3.99-0.5                  fs_1.5.0                     
## [145] splines_4.0.3                 RBGL_1.66.0                  
## [147] statmod_1.4.35                sn_1.6-2                     
## [149] expm_0.999-6                  pkgdown_1.6.1                
## [151] graphlayouts_0.7.1            multtest_2.46.0              
## [153] flexmix_2.3-17                systemfonts_0.3.2            
## [155] xtable_1.8-4                  jsonlite_1.7.2               
## [157] tidygraph_1.2.0               corpcor_1.6.9                
## [159] modeltools_0.2-23             R6_2.5.0                     
## [161] gmodels_2.18.1                TFisher_0.2.0                
## [163] pillar_1.4.7                  htmltools_0.5.1.1            
## [165] mime_0.9                      glue_1.4.2                   
## [167] fastmap_1.1.0                 BiocParallel_1.24.1          
## [169] class_7.3-17                  interactiveDisplayBase_1.28.0
## [171] codetools_0.2-16              tsne_0.1-3                   
## [173] mvtnorm_1.1-1                 lattice_0.20-41              
## [175] logger_0.1                    numDeriv_2016.8-1.1          
## [177] curl_4.3                      gtools_3.8.2                 
## [179] openssl_1.4.3                 survival_3.2-7               
## [181] limma_3.46.0                  rmarkdown_2.6                
## [183] fastICA_1.2-2                 desc_1.2.0                   
## [185] munsell_0.5.0                 e1071_1.7-4                  
## [187] fastcluster_1.1.25            GenomeInfoDbData_1.2.4       
## [189] reshape2_1.4.4                haven_2.3.1                  
## [191] gtable_0.3.0                  rbibutils_2.0</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-hanahan2011hallmarks">
<p>Hanahan, Douglas, and Robert A Weinberg. 2011. “Hallmarks of Cancer: The Next Generation.” <em>Cell</em> 144 (5): 646–74.</p>
</div>
<div id="ref-kanehisa2004kegg">
<p>Kanehisa, Minoru, Susumu Goto, Shuichi Kawashima, Yasushi Okuno, and Masahiro Hattori. 2004. “The Kegg Resource for Deciphering the Genome.” <em>Nucleic Acids Research</em> 32 (suppl_1): D277–D280.</p>
</div>
<div id="ref-pearl2010causal">
<p>Pearl, Judea. 2010. “Causal Inference.” <em>Causality: Objectives and Assessment</em>, 39–58.</p>
</div>
<div id="ref-tomczak2015cancer">
<p>Tomczak, Katarzyna, Patrycja Czerwińska, and Maciej Wiznerowicz. 2015. “The Cancer Genome Atlas (Tcga): An Immeasurable Source of Knowledge.” <em>Contemporary Oncology</em> 19 (1A): A68.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Kim Philipp Jablonski, Martin Pirkl.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
