<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Overview of pathway network databases • dce</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Overview of pathway network databases">
<meta property="og:description" content="dce">
<meta property="og:image" content="https://cbg-ethz.github.io/dce/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">dce</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/dce.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/pathway_databases.html">Overview of pathway network databases</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/cbg-ethz/dce/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="pathway_databases_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Overview of pathway network databases</h1>
                        <h4 data-toc-skip class="author">Kim Philipp Jablonski, Martin Pirkl</h4>
            
            <h4 data-toc-skip class="date">2022-03-30</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/cbg-ethz/dce/blob/HEAD/vignettes/pathway_databases.Rmd" class="external-link"><code>vignettes/pathway_databases.Rmd</code></a></small>
      <div class="hidden name"><code>pathway_databases.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<div class="section level3">
<h3 id="load-required-packages">Load required packages<a class="anchor" aria-label="anchor" href="#load-required-packages"></a>
</h3>
<p>Load the package with the library function.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/cbg-ethz/dce" class="external-link">dce</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="pathway-database-overview">Pathway database overview<a class="anchor" aria-label="anchor" href="#pathway-database-overview"></a>
</h2>
<p>We provide access to the following topological pathway databases using graphite <span class="citation">(Sales et al. 2012)</span> in a processed format. This format looks as follows:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">dce</span><span class="fu">::</span><span class="va"><a href="../reference/df_pathway_statistics.html">df_pathway_statistics</a></span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">node_num</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">database</th>
<th align="left">pathway_id</th>
<th align="left">pathway_name</th>
<th align="right">node_num</th>
<th align="right">edge_num</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">R-HSA-162582</td>
<td align="left">Signaling Pathways</td>
<td align="right">2488</td>
<td align="right">62068</td>
</tr>
<tr class="even">
<td align="left">reactome</td>
<td align="left">R-HSA-1430728</td>
<td align="left">Metabolism</td>
<td align="right">2047</td>
<td align="right">85543</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">R-HSA-392499</td>
<td align="left">Metabolism of proteins</td>
<td align="right">1894</td>
<td align="right">52807</td>
</tr>
<tr class="even">
<td align="left">reactome</td>
<td align="left">R-HSA-1643685</td>
<td align="left">Disease</td>
<td align="right">1774</td>
<td align="right">55469</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">R-HSA-168256</td>
<td align="left">Immune System</td>
<td align="right">1771</td>
<td align="right">58277</td>
</tr>
<tr class="even">
<td align="left">panther</td>
<td align="left">P00057</td>
<td align="left">Wnt signaling pathway</td>
<td align="right">1644</td>
<td align="right">195344</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">R-HSA-74160</td>
<td align="left">Gene expression (Transcription)</td>
<td align="right">1472</td>
<td align="right">32493</td>
</tr>
<tr class="even">
<td align="left">reactome</td>
<td align="left">R-HSA-597592</td>
<td align="left">Post-translational protein modification</td>
<td align="right">1394</td>
<td align="right">26399</td>
</tr>
<tr class="odd">
<td align="left">kegg</td>
<td align="left">hsa:01100</td>
<td align="left">Metabolic pathways</td>
<td align="right">1343</td>
<td align="right">22504</td>
</tr>
<tr class="even">
<td align="left">reactome</td>
<td align="left">R-HSA-73857</td>
<td align="left">RNA Polymerase II Transcription</td>
<td align="right">1339</td>
<td align="right">25294</td>
</tr>
</tbody>
</table>
<p>Let’s see how many pathways each database provides:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">dce</span><span class="fu">::</span><span class="va"><a href="../reference/df_pathway_statistics.html">df_pathway_statistics</a></span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">database</span>, sort <span class="op">=</span> <span class="cn">TRUE</span>, name <span class="op">=</span> <span class="st">"pathway_number"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">database</th>
<th align="right">pathway_number</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">pathbank</td>
<td align="right">48685</td>
</tr>
<tr class="even">
<td align="left">smpdb</td>
<td align="right">48671</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="right">2406</td>
</tr>
<tr class="even">
<td align="left">wikipathways</td>
<td align="right">640</td>
</tr>
<tr class="odd">
<td align="left">kegg</td>
<td align="right">323</td>
</tr>
<tr class="even">
<td align="left">panther</td>
<td align="right">94</td>
</tr>
<tr class="odd">
<td align="left">pharmgkb</td>
<td align="right">90</td>
</tr>
</tbody>
</table>
<p>Next, we can see how the pathway sizes are distributed for each database:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">dce</span><span class="fu">::</span><span class="va"><a href="../reference/df_pathway_statistics.html">df_pathway_statistics</a></span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">node_num</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_histogram</a></span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">30</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">database</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="pathway_databases_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="plotting-pathways">Plotting pathways<a class="anchor" aria-label="anchor" href="#plotting-pathways"></a>
</h2>
<p>It is easily possible to plot pathways:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_pathways.html">get_pathways</a></span><span class="op">(</span>
  pathway_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
    pathbank <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Lactose Synthesis"</span><span class="op">)</span>,
    kegg <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Fatty acid biosynthesis"</span><span class="op">)</span>
  <span class="op">)</span>
<span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">pathways</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="../reference/plot_network.html">plot_network</a></span><span class="op">(</span>
    <span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">graph</span>, <span class="st">"matrix"</span><span class="op">)</span>,
    visualize_edge_weights <span class="op">=</span> <span class="cn">FALSE</span>,
    arrow_size <span class="op">=</span> <span class="fl">0.02</span>,
    shadowtext <span class="op">=</span> <span class="cn">TRUE</span>
  <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">pathway_name</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [[1]]</span></code></pre>
<p><img src="pathway_databases_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<pre><code><span class="co">## </span>
<span class="co">## [[2]]</span></code></pre>
<p><img src="pathway_databases_files/figure-html/unnamed-chunk-5-2.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.3 (2022-03-10)</span>
<span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">## Running under: macOS Big Sur/Monterey 10.16</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib</span>
<span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] org.Hs.eg.db_3.14.0  AnnotationDbi_1.56.2 IRanges_2.28.0      </span>
<span class="co">##  [4] S4Vectors_0.32.4     Biobase_2.54.0       BiocGenerics_0.40.0 </span>
<span class="co">##  [7] dce_0.99.6           forcats_0.5.1        stringr_1.4.0       </span>
<span class="co">## [10] dplyr_1.0.8          purrr_0.3.4          readr_2.1.2         </span>
<span class="co">## [13] tidyr_1.2.0          tibble_3.1.6         ggplot2_3.3.5       </span>
<span class="co">## [16] tidyverse_1.3.1      BiocStyle_2.22.0    </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] rappdirs_0.3.3           prabclus_2.3-2           ragg_1.2.2              </span>
<span class="co">##   [4] bit64_4.0.5              knitr_1.38               multcomp_1.4-18         </span>
<span class="co">##   [7] wesanderson_0.3.6        data.table_1.14.2        KEGGREST_1.34.0         </span>
<span class="co">##  [10] RCurl_1.98-1.6           generics_0.1.2           metap_1.8               </span>
<span class="co">##  [13] TH.data_1.1-0            RSQLite_2.2.11           shadowtext_0.1.1        </span>
<span class="co">##  [16] proxy_0.4-26             CombinePValue_1.0        bit_4.0.4               </span>
<span class="co">##  [19] tzdb_0.3.0               mutoss_0.1-12            xml2_1.3.3              </span>
<span class="co">##  [22] lubridate_1.8.0          assertthat_0.2.1         viridis_0.6.2           </span>
<span class="co">##  [25] amap_0.8-18              xfun_0.30                hms_1.1.1               </span>
<span class="co">##  [28] jquerylib_0.1.4          evaluate_0.15            DEoptimR_1.0-10         </span>
<span class="co">##  [31] fansi_1.0.3              dbplyr_2.1.1             readxl_1.4.0            </span>
<span class="co">##  [34] Rgraphviz_2.38.0         igraph_1.2.11            DBI_1.1.2               </span>
<span class="co">##  [37] tmvnsim_1.0-2            apcluster_1.4.9          RcppArmadillo_0.10.8.1.0</span>
<span class="co">##  [40] ellipsis_0.3.2           backports_1.4.1          bookdown_0.25           </span>
<span class="co">##  [43] permute_0.9-7            harmonicmeanp_3.0        vctrs_0.4.0             </span>
<span class="co">##  [46] Linnorm_2.18.0           abind_1.4-5              cachem_1.0.6            </span>
<span class="co">##  [49] RcppEigen_0.3.3.9.1      withr_2.5.0              ggforce_0.3.3           </span>
<span class="co">##  [52] sfsmisc_1.1-12           robustbase_0.93-9        bdsmatrix_1.3-4         </span>
<span class="co">##  [55] checkmate_2.0.0          vegan_2.5-7              pcalg_2.7-5             </span>
<span class="co">##  [58] mclust_5.4.9             mnormt_2.0.2             cluster_2.1.2           </span>
<span class="co">##  [61] crayon_1.5.1             ellipse_0.4.2            FMStable_0.1-2          </span>
<span class="co">##  [64] edgeR_3.36.0             pkgconfig_2.0.3          labeling_0.4.2          </span>
<span class="co">##  [67] qqconf_1.2.3             tweenr_1.0.2             GenomeInfoDb_1.30.1     </span>
<span class="co">##  [70] nlme_3.1-155             ggm_2.5                  nnet_7.3-17             </span>
<span class="co">##  [73] rlang_1.0.2              diptest_0.76-0           lifecycle_1.0.1         </span>
<span class="co">##  [76] sandwich_3.0-1           mathjaxr_1.6-0           modelr_0.1.8            </span>
<span class="co">##  [79] cellranger_1.1.0         rprojroot_2.0.2          polyclip_1.10-0         </span>
<span class="co">##  [82] matrixStats_0.61.0       graph_1.72.0             Matrix_1.4-0            </span>
<span class="co">##  [85] zoo_1.8-9                reprex_2.0.1             png_0.1-7               </span>
<span class="co">##  [88] viridisLite_0.4.0        bitops_1.0-7             Biostrings_2.62.0       </span>
<span class="co">##  [91] blob_1.2.2               scales_1.1.1             memoise_2.0.1           </span>
<span class="co">##  [94] graphite_1.40.0          magrittr_2.0.3           plyr_1.8.7              </span>
<span class="co">##  [97] gdata_2.18.0             zlibbioc_1.40.0          compiler_4.1.3          </span>
<span class="co">## [100] plotrix_3.8-2            clue_0.3-60              cli_3.2.0               </span>
<span class="co">## [103] XVector_0.34.0           MASS_7.3-55              mgcv_1.8-39             </span>
<span class="co">## [106] tidyselect_1.1.2         stringi_1.7.6            textshaping_0.3.6       </span>
<span class="co">## [109] highr_0.9                yaml_2.3.5               locfit_1.5-9.5          </span>
<span class="co">## [112] ggrepel_0.9.1            grid_4.1.3               sass_0.4.1              </span>
<span class="co">## [115] tools_4.1.3              parallel_4.1.3           rstudioapi_0.13         </span>
<span class="co">## [118] snowfall_1.84-6.1        gridExtra_2.3            farver_2.1.0            </span>
<span class="co">## [121] Rtsne_0.15               ggraph_2.0.5             digest_0.6.29           </span>
<span class="co">## [124] BiocManager_1.30.16      flexclust_1.4-0          mnem_1.10.0             </span>
<span class="co">## [127] fpc_2.2-9                ppcor_1.1                Rcpp_1.0.8.3            </span>
<span class="co">## [130] broom_0.7.12             httr_1.4.2               ggdendro_0.1.23         </span>
<span class="co">## [133] kernlab_0.9-29           naturalsort_0.1.3        Rdpack_2.3              </span>
<span class="co">## [136] colorspace_2.0-3         rvest_1.0.2              fs_1.5.2                </span>
<span class="co">## [139] splines_4.1.3            RBGL_1.70.0              statmod_1.4.36          </span>
<span class="co">## [142] sn_2.0.2                 expm_0.999-6             pkgdown_2.0.2           </span>
<span class="co">## [145] graphlayouts_0.8.0       multtest_2.50.0          flexmix_2.3-17          </span>
<span class="co">## [148] systemfonts_1.0.4        jsonlite_1.8.0           tidygraph_1.2.0         </span>
<span class="co">## [151] corpcor_1.6.10           modeltools_0.2-23        R6_2.5.1                </span>
<span class="co">## [154] gmodels_2.18.1           TFisher_0.2.0            pillar_1.7.0            </span>
<span class="co">## [157] htmltools_0.5.2          glue_1.6.2               fastmap_1.1.0           </span>
<span class="co">## [160] class_7.3-20             codetools_0.2-18         tsne_0.1-3.1            </span>
<span class="co">## [163] mvtnorm_1.1-3            utf8_1.2.2               lattice_0.20-45         </span>
<span class="co">## [166] bslib_0.3.1              logger_0.2.2             numDeriv_2016.8-1.1     </span>
<span class="co">## [169] curl_4.3.2               gtools_3.9.2             survival_3.2-13         </span>
<span class="co">## [172] limma_3.50.1             rmarkdown_2.13           desc_1.4.1              </span>
<span class="co">## [175] fastICA_1.2-3            munsell_0.5.0            e1071_1.7-9             </span>
<span class="co">## [178] fastcluster_1.2.3        GenomeInfoDbData_1.2.7   haven_2.4.3             </span>
<span class="co">## [181] reshape2_1.4.4           gtable_0.3.0             rbibutils_2.2.7</span></code></pre>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references">
<div id="ref-sales2012graphite">
<p>Sales, Gabriele, Enrica Calura, Duccio Cavalieri, and Chiara Romualdi. 2012. “Graphite-a Bioconductor Package to Convert Pathway Topology to Gene Network.” <em>BMC Bioinformatics</em> 13 (1): 20.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Kim Philipp Jablonski, Martin Pirkl.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
