<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Overview of pathway network databases • dce</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Overview of pathway network databases">
<meta property="og:description" content="dce">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">dce</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/dce.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/pathway_databases.html">Overview of pathway network databases</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/cbg-ethz/dce/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="pathway_databases_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Overview of pathway network databases</h1>
                        <h4 class="author">Kim Philipp Jablonski, Martin Pirkl</h4>
            
            <h4 class="date">2021-08-03</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/cbg-ethz/dce/blob/master/vignettes/pathway_databases.Rmd"><code>vignettes/pathway_databases.Rmd</code></a></small>
      <div class="hidden name"><code>pathway_databases.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<div id="load-required-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#load-required-packages" class="anchor"></a>Load required packages</h2>
<p>Load the package with the library function.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/cbg-ethz/dce">dce</a></span><span class="op">)</span></code></pre></div>
<pre><code>## Warning: replacing previous import 'stats::filter' by 'dplyr::filter' when
## loading 'dce'</code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="pathway-database-overview" class="section level1">
<h1 class="hasAnchor">
<a href="#pathway-database-overview" class="anchor"></a>Pathway database overview</h1>
<p>We provide access to the following topological pathway databases using graphite <span class="citation">(Sales et al. 2012)</span> in a processed format. This format looks as follows:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">dce</span><span class="fu">::</span><span class="va"><a href="../reference/df_pathway_statistics.html">df_pathway_statistics</a></span> <span class="op">%&gt;%</span>
  <span class="fu">arrange</span><span class="op">(</span><span class="fu">desc</span><span class="op">(</span><span class="va">node_num</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">database</th>
<th align="left">pathway_id</th>
<th align="left">pathway_name</th>
<th align="right">node_num</th>
<th align="right">edge_num</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">R-HSA-162582</td>
<td align="left">Signaling Pathways</td>
<td align="right">2479</td>
<td align="right">61904</td>
</tr>
<tr class="even">
<td align="left">reactome</td>
<td align="left">R-HSA-1430728</td>
<td align="left">Metabolism</td>
<td align="right">2047</td>
<td align="right">82767</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">R-HSA-392499</td>
<td align="left">Metabolism of proteins</td>
<td align="right">1962</td>
<td align="right">53074</td>
</tr>
<tr class="even">
<td align="left">reactome</td>
<td align="left">R-HSA-168256</td>
<td align="left">Immune System</td>
<td align="right">1771</td>
<td align="right">58241</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">R-HSA-1643685</td>
<td align="left">Disease</td>
<td align="right">1730</td>
<td align="right">54865</td>
</tr>
<tr class="even">
<td align="left">panther</td>
<td align="left">P00057</td>
<td align="left">Wnt signaling pathway</td>
<td align="right">1644</td>
<td align="right">195344</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">R-HSA-74160</td>
<td align="left">Gene expression (Transcription)</td>
<td align="right">1472</td>
<td align="right">32476</td>
</tr>
<tr class="even">
<td align="left">reactome</td>
<td align="left">R-HSA-597592</td>
<td align="left">Post-translational protein modification</td>
<td align="right">1393</td>
<td align="right">26392</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">R-HSA-73857</td>
<td align="left">RNA Polymerase II Transcription</td>
<td align="right">1339</td>
<td align="right">25277</td>
</tr>
<tr class="even">
<td align="left">kegg</td>
<td align="left">hsa:01100</td>
<td align="left">Metabolic pathways</td>
<td align="right">1314</td>
<td align="right">21552</td>
</tr>
</tbody>
</table>
<p>Let’s see how many pathways each database provides:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">dce</span><span class="fu">::</span><span class="va"><a href="../reference/df_pathway_statistics.html">df_pathway_statistics</a></span> <span class="op">%&gt;%</span>
  <span class="fu">count</span><span class="op">(</span><span class="va">database</span>, sort <span class="op">=</span> <span class="cn">TRUE</span>, name <span class="op">=</span> <span class="st">"pathway_number"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">database</th>
<th align="right">pathway_number</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">pathbank</td>
<td align="right">48685</td>
</tr>
<tr class="even">
<td align="left">smpdb</td>
<td align="right">48671</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="right">2390</td>
</tr>
<tr class="even">
<td align="left">wikipathways</td>
<td align="right">601</td>
</tr>
<tr class="odd">
<td align="left">kegg</td>
<td align="right">321</td>
</tr>
<tr class="even">
<td align="left">panther</td>
<td align="right">94</td>
</tr>
<tr class="odd">
<td align="left">pharmgkb</td>
<td align="right">90</td>
</tr>
</tbody>
</table>
<p>Next, we can see how the pathway sizes are distributed for each database:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">dce</span><span class="fu">::</span><span class="va"><a href="../reference/df_pathway_statistics.html">df_pathway_statistics</a></span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">node_num</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html">geom_histogram</a></span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">30</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">database</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="pathway_databases_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
</div>
<div id="plotting-pathways" class="section level1">
<h1 class="hasAnchor">
<a href="#plotting-pathways" class="anchor"></a>Plotting pathways</h1>
<p>It is easily possible to plot pathways:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_pathways.html">get_pathways</a></span><span class="op">(</span>
  pathway_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>
    pathbank <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Lactose Synthesis"</span><span class="op">)</span>,
    kegg <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Fatty acid biosynthesis"</span><span class="op">)</span>,
    wikipathways <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"TGF-beta Receptor Signaling"</span><span class="op">)</span>
  <span class="op">)</span>
<span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">pathways</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="../reference/plot_network.html">plot_network</a></span><span class="op">(</span>
    <span class="fu">as</span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">graph</span>, <span class="st">"matrix"</span><span class="op">)</span>,
    visualize_edge_weights <span class="op">=</span> <span class="cn">FALSE</span>,
    arrow_size <span class="op">=</span> <span class="fl">0.02</span>,
    shadowtext <span class="op">=</span> <span class="cn">TRUE</span>
  <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">pathway_name</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<pre><code>## [[1]]</code></pre>
<p><img src="pathway_databases_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<pre><code>## 
## [[2]]</code></pre>
<p><img src="pathway_databases_files/figure-html/unnamed-chunk-5-2.png" width="700"></p>
<pre><code>## 
## [[3]]</code></pre>
<p><img src="pathway_databases_files/figure-html/unnamed-chunk-5-3.png" width="700"></p>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.0 (2021-05-18)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Catalina 10.15.7
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] org.Hs.eg.db_3.13.0  AnnotationDbi_1.54.1 IRanges_2.26.0      
##  [4] S4Vectors_0.30.0     Biobase_2.52.0       BiocGenerics_0.38.0 
##  [7] dce_0.99.6           forcats_0.5.1        stringr_1.4.0       
## [10] dplyr_1.0.7          purrr_0.3.4          readr_2.0.0         
## [13] tidyr_1.1.3          tibble_3.1.3         ggplot2_3.3.5       
## [16] tidyverse_1.3.1      BiocStyle_2.20.2    
## 
## loaded via a namespace (and not attached):
##   [1] rappdirs_0.3.3           prabclus_2.3-2           ragg_1.1.3              
##   [4] bit64_4.0.5              knitr_1.33               multcomp_1.4-17         
##   [7] wesanderson_0.3.6        data.table_1.14.0        KEGGREST_1.32.0         
##  [10] RCurl_1.98-1.3           generics_0.1.0           metap_1.4               
##  [13] TH.data_1.0-10           RSQLite_2.2.7            shadowtext_0.0.8        
##  [16] proxy_0.4-26             CombinePValue_1.0        bit_4.0.4               
##  [19] tzdb_0.1.2               mutoss_0.1-12            xml2_1.3.2              
##  [22] lubridate_1.7.10         assertthat_0.2.1         viridis_0.6.1           
##  [25] amap_0.8-18              xfun_0.24                hms_1.1.0               
##  [28] jquerylib_0.1.4          evaluate_0.14            DEoptimR_1.0-9          
##  [31] fansi_0.5.0              dbplyr_2.1.1             readxl_1.3.1            
##  [34] Rgraphviz_2.36.0         igraph_1.2.6             DBI_1.1.1               
##  [37] tmvnsim_1.0-2            apcluster_1.4.8          RcppArmadillo_0.10.6.0.0
##  [40] ellipsis_0.3.2           backports_1.2.1          bookdown_0.22           
##  [43] permute_0.9-5            harmonicmeanp_3.0        vctrs_0.3.8             
##  [46] Linnorm_2.16.0           abind_1.4-5              cachem_1.0.5            
##  [49] RcppEigen_0.3.3.9.1      withr_2.4.2              ggforce_0.3.3           
##  [52] sfsmisc_1.1-11           robustbase_0.93-8        bdsmatrix_1.3-4         
##  [55] checkmate_2.0.0          vegan_2.5-7              pcalg_2.7-3             
##  [58] mclust_5.4.7             mnormt_2.0.2             cluster_2.1.2           
##  [61] crayon_1.4.1             ellipse_0.4.2            FMStable_0.1-2          
##  [64] edgeR_3.34.0             pkgconfig_2.0.3          labeling_0.4.2          
##  [67] tweenr_1.0.2             GenomeInfoDb_1.28.1      nlme_3.1-152            
##  [70] ggm_2.5                  nnet_7.3-16              rlang_0.4.11            
##  [73] diptest_0.76-0           lifecycle_1.0.0          sandwich_3.0-1          
##  [76] mathjaxr_1.4-0           modelr_0.1.8             cellranger_1.1.0        
##  [79] rprojroot_2.0.2          polyclip_1.10-0          matrixStats_0.60.0      
##  [82] graph_1.70.0             Matrix_1.3-3             zoo_1.8-9               
##  [85] reprex_2.0.0             png_0.1-7                viridisLite_0.4.0       
##  [88] bitops_1.0-7             Biostrings_2.60.1        blob_1.2.2              
##  [91] scales_1.1.1             memoise_2.0.0            graphite_1.38.0         
##  [94] magrittr_2.0.1           plyr_1.8.6               gdata_2.18.0            
##  [97] zlibbioc_1.38.0          compiler_4.1.0           plotrix_3.8-1           
## [100] clue_0.3-59              cli_3.0.1                XVector_0.32.0          
## [103] MASS_7.3-54              mgcv_1.8-35              tidyselect_1.1.1        
## [106] stringi_1.7.3            textshaping_0.3.5        highr_0.9               
## [109] yaml_2.2.1               locfit_1.5-9.4           ggrepel_0.9.1           
## [112] grid_4.1.0               sass_0.4.0               tools_4.1.0             
## [115] rstudioapi_0.13          snowfall_1.84-6.1        gridExtra_2.3           
## [118] farver_2.1.0             Rtsne_0.15               ggraph_2.0.5            
## [121] digest_0.6.27            BiocManager_1.30.16      flexclust_1.4-0         
## [124] mnem_1.8.0               fpc_2.2-9                ppcor_1.1               
## [127] Rcpp_1.0.7               broom_0.7.9              httr_1.4.2              
## [130] ggdendro_0.1.22          kernlab_0.9-29           naturalsort_0.1.3       
## [133] Rdpack_2.1.2             colorspace_2.0-2         rvest_1.0.1             
## [136] fs_1.5.0                 splines_4.1.0            RBGL_1.68.0             
## [139] statmod_1.4.36           sn_2.0.0                 expm_0.999-6            
## [142] pkgdown_1.6.1            graphlayouts_0.7.1       multtest_2.48.0         
## [145] flexmix_2.3-17           systemfonts_1.0.2        jsonlite_1.7.2          
## [148] tidygraph_1.2.0          corpcor_1.6.9            modeltools_0.2-23       
## [151] R6_2.5.0                 gmodels_2.18.1           TFisher_0.2.0           
## [154] pillar_1.6.2             htmltools_0.5.1.1        glue_1.4.2              
## [157] fastmap_1.1.0            class_7.3-19             codetools_0.2-18        
## [160] tsne_0.1-3               mvtnorm_1.1-2            utf8_1.2.2              
## [163] lattice_0.20-44          bslib_0.2.5.1            logger_0.2.1            
## [166] numDeriv_2016.8-1.1      curl_4.3.2               gtools_3.9.2            
## [169] survival_3.2-11          limma_3.48.1             rmarkdown_2.9           
## [172] desc_1.3.0               fastICA_1.2-2            munsell_0.5.0           
## [175] e1071_1.7-8              fastcluster_1.2.3        GenomeInfoDbData_1.2.6  
## [178] haven_2.4.2              reshape2_1.4.4           gtable_0.3.0            
## [181] rbibutils_2.2.2</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-sales2012graphite">
<p>Sales, Gabriele, Enrica Calura, Duccio Cavalieri, and Chiara Romualdi. 2012. “Graphite-a Bioconductor Package to Convert Pathway Topology to Gene Network.” <em>BMC Bioinformatics</em> 13 (1): 20.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Kim Philipp Jablonski, Martin Pirkl.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
